components/camera/videosource/webcam.go

```diff
@@ -217,48 +216,125 @@ func makeConstraints(conf *WebcamConfig, debug bool, logger logging.Logger) medi
 	}
 }
 
+// getNamedVideoSource is a helper function for trying to open a webcam that attempts to
+// find a video device (not a screen) by the given name.
+//
+// First it will try to use the path name after evaluating any symbolic links. If
+// evaluation fails, it will try to use the path name as provided.
+func getNamedVideoSource(
+	path string,
+	fromLabel bool,
+	constraints mediadevices.MediaStreamConstraints,
+	logger logging.Logger,
+) (gostream.MediaSource[image.Image], error) {
+	if !fromLabel {
+		resolvedPath, err := filepath.EvalSymlinks(path)
+		if err == nil {
+			path = resolvedPath
+		}
+	}
+	return gostream.GetNamedVideoSource(filepath.Base(path), constraints, logger)
+}
+
+// tryWebcamOpen is a helper for finding and making the video source that uses getNamedVideoSource to try and find
+// a video device (gostream.MediaSource). If successful, it will wrap that MediaSource in a VideoSource.
+func tryWebcamOpen(
+	ctx context.Context,
+	conf *WebcamConfig,
+	path string,
+	fromLabel bool,
+	constraints mediadevices.MediaStreamConstraints,
+	logger logging.Logger,
+) (gostream.VideoSource, error) {
+	source, err := getNamedVideoSource(path, fromLabel, constraints, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	if conf.Width != 0 && conf.Height != 0 {
+		img, release, err := gostream.ReadMedia(ctx, source)
+		if release != nil {
+			defer release()
+		}
+		if err != nil {
+			return nil, err
+		}
+		if img.Bounds().Dx() != conf.Width || img.Bounds().Dy() != conf.Height {
+			return nil, errors.Errorf("requested width and height (%dx%d) are not available for this webcam"+
```

Comment:
Should we fail here or just waning log?

Reply:
Warning log + what behavior though

Reply:
If I am reading this correctly, it is forcing an error if the driver and upstream ReadMedia call is returning different width and height from what is requested in the conf. I like the idea of being more lenient here but prob does make sense to change behavior in a cleanup PR.

Reply:
ok noted for future thanks for callout